app/subscriber/src/features/settings/MyAccountSettings.tsx

```
@@ -7,6 +7,38 @@ import { ISubscriberUserModel, ToggleButton } from 'tno-core';
 
 import * as styled from './styled';
 
+export const toggleVacationMode = async (
+  profile: ISubscriberUserModel | undefined,
+  impersonate: ISubscriberUserModel | undefined,
+  isVacationMode: boolean,
+  updateUser: (model: ISubscriberUserModel, impersonate?: boolean) => Promise<ISubscriberUserModel>,
+) => {
+  if (!profile) {
+    toast.error('User information is missing. Please try again later');
+    return;
+  }
+  const baseProfile = impersonate ?? profile;
+  const createUser = (): ISubscriberUserModel => {
+    // use impersonate if it exists, otherwise use profile
+    return {
+      ...baseProfile,
+      preferences: {
+        ...baseProfile.preferences,
+        isVacationMode: isVacationMode,
+      },
+    };
+  };
+  const user = createUser();
+
+  try {
+    await updateUser(user, !!impersonate);
+    toast.success('Vacation mode has successfully been updated.');
+  } catch (error) {
+    // Handle the error, if needed
+    console.error('Failed to update user:', error);
```

Any reason a failure is hidden from the user?